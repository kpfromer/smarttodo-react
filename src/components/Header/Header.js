import React from "react";
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { NavLink } from "react-router-dom";
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

import styles from './Header.module.css';

export const Header = ({ isAuthenticated }) => {

  const menuItems =
    isAuthenticated ?
      <div id="logged-in-links">
        <NavLink to="/logout">
          <Button color="inherit">
            Logout
          </Button>
        </NavLink>
      </div>
      :
      <div id="logged-out-links">
        <NavLink to="/register">
          <Button color="inherit">
            Register
          </Button>
        </NavLink>
        <NavLink to="/login">
          <Button color="inherit">
            Login
          </Button>
        </NavLink>
      </div>;

  return (
    <AppBar position="static">
      <Toolbar>
        <IconButton className={styles.menuButton} color="inherit" aria-label="Menu">
          <MenuIcon />
        </IconButton>
        <Typography variant="title" color="inherit" className={styles.flex}>
          <NavLink to="/">Smarttodo</NavLink>
        </Typography>
        {menuItems}
      </Toolbar>
    </AppBar>
  )
};

Header.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired
};

export const mapStateToProps = state => ({
  isAuthenticated: !!state.login.token
});

export default connect(mapStateToProps)(Header);