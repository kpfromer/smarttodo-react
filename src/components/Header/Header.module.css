.mainContainer {
  background-color: #3573E7;
}

.title {
  font-size: 1.35em;
  margin: 0;
}

.navList {
  margin-top: 5px;
}

.title a,
.navList a {
  text-align: center;
  display: block;
  padding: 10px 15px;
}

.navList a {
  font-size: .95em;
  text-transform: uppercase;
}

@media (min-width: 769px) {
  .navContainer,
  .navList {
    display: flex;
  }

  .navContainer {
    flex-direction: column;
    align-items: center;
  }
}

@media (min-width: 1125px) {
  .navContainer,
  .navList {
    flex-direction: row;
  }
  .navList {
    margin-left: auto;
  }
}

.flex {
  flex: 1;
}

.menuButton {
  margin-left: -12px;
  margin-right: 20px;
}