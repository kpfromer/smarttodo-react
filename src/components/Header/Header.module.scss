@import '../../theme';

.mainContainer {
  background-color: $main-color;
}

.title {
  font-size: 1.35em;
  margin: 0;
}

.navList {
  margin-top: 5px;
}

.title a,
.navList a {
  text-align: center;
  display: block;
  padding: 10px 15px;
}

.navList a {
  font-size: .95em;
  text-transform: uppercase;
}

@media (min-width: $sm) {
  .navContainer,
  .navList {
    display: flex;
  }

  .navContainer {
    flex-direction: column;
    align-items: center;
  }
}

@media (min-width: $md) {
  .navContainer,
  .navList {
    flex-direction: row;
  }
  .navList {
    margin-left: auto;
  }
}

.flex {
  flex: 1;
}

.menuButton {
  margin-left: -12px;
  margin-right: 20px;
}