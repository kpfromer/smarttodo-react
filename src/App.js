import React, { Component, Fragment } from 'react';
import { BrowserRouter, Route, Switch } from "react-router-dom";
//styles
import './App.scss';

import { Header } from "./components/Header/Header";
import TodoList from "./containers/TodoList/TodoList";
import NotFound from "./components/NotFound/NotFound";
import Logout from "./components/Logout/Logout";


class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <Fragment>
          <Header/>
          <div className="container">
            <Switch>
              <Route exact path="/" component={TodoList} />
              <Route exact path="/logout" component={Logout} />
              <Route component={NotFound}/>
            </Switch>
          </div>
        </Fragment>
      </BrowserRouter>
    );
  }
}

export default App;
