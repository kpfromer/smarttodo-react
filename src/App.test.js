import React from 'react';
import App from './App';
import { SetupComponent } from "react-component-setup";
import Header from "./components/Header/Header";
import { Switch } from "react-router-dom";

const { shallow: setup } = SetupComponent({
  Component: App
});

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    ({ wrapper } = setup());
  });

  it('renders without crashing', () => {
    setup();
  });

  it('renders a header', () => {
    expect(wrapper.find(Header)).toExist();
  });

  it('renders routes', () => {
    expect(wrapper.find(Switch)).toMatchSnapshot();
  });
});