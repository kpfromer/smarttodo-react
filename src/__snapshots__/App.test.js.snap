// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders routes 1`] = `
<Switch>
  <Connect(PrivateRoute)
    component={[Function]}
    path="/todo"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/logout"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
